tesseract_time_parameterization/include/tesseract_time_parameterization/time_optimal_trajectory_generation.h

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2011-2012, Georgia Tech Research Corporation
+ * All rights reserved.
+ *
+ * Author: Tobias Kunz <tobias@gatech.edu>
+ * Date: 05/2012
+ *
+ * Humanoid Robotics Lab      Georgia Institute of Technology
+ * Director: Mike Stilman     http://www.golems.org
+ *
+ * Algorithm details and publications:
+ * http://www.golems.org/node/1570
+ *
+ * This file is provided under the following "BSD-style" License:
+ *   Redistribution and use in source and binary forms, with or
+ *   without modification, are permitted provided that the following
+ *   conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+ *   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ *   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *   LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+ *   USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *   AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *   LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *   ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef TESSERACT_TIME_PARAMETERIZATION_TIME_OPTIMAL_TRAJECTORY_GENERATION_H
+#define TESSERACT_TIME_PARAMETERIZATION_TIME_OPTIMAL_TRAJECTORY_GENERATION_H
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <Eigen/Eigen>
+#include <list>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_command_language/command_language.h>
+
+namespace tesseract_planning
+{
+class TimeOptimalTrajectoryGeneration
+{
+public:
+  TimeOptimalTrajectoryGeneration(double path_tolerance = 0.1,
+                                  double resample_dt = 0.1,
+                                  double min_angle_change = 0.001);
+
+  bool computeTimeStamps(CompositeInstruction& program,
+                         const Eigen::Ref<const Eigen::VectorXd>& max_velocity,
+                         const Eigen::Ref<const Eigen::VectorXd>& max_acceleration,
+                         double max_velocity_scaling_factor = 1.0,
+                         double max_acceleration_scaling_factor = 1.0) const;
+
+private:
+  const double path_tolerance_;
+  const double resample_dt_;
+  const double min_angle_change_;
+};
+}  // namespace tesseract_planning
+
+namespace totg
```

Comment:
Should we just encapsulate this under the tesseract_planning namespace and remove totg?

Reply:
I actually added that namespace because I thought Trajectory, Path, etc were pretty generic, and I didn't want there to be clashes. A user should never really touch those anyway.

Reply:
Makes sense. It might be good move togo namespace inside the tesseact_planning namespace?

Reply:
Sure. I can do that.